src/WebJobs.Script.WebHost/WebScriptHostHttpRoutesManager.cs

```diff
@@ -34,7 +37,14 @@ public void InitializeHttpFunctionRoutes(IScriptJobHost host)
             _router.ClearRoutes();
 
             // TODO: FACAVAL Instantiation of the ScriptRouteHandler should be cleaned up
-            WebJobsRouteBuilder routesBuilder = _router.CreateBuilder(new ScriptRouteHandler(_loggerFactory, host, _environment, false), _httpOptions.Value.RoutePrefix);
+            // If routeHandling.mode is "all", map routes at root (no routePrefix). Otherwise honor configured routePrefix.
+            string routePrefixToUse = _httpOptions.Value.RoutePrefix;
+            if (string.Equals(_routeHandlingOptions?.Mode, "all", StringComparison.OrdinalIgnoreCase))
+            {
+                routePrefixToUse = null;
+            }
+
+            WebJobsRouteBuilder routesBuilder = _router.CreateBuilder(new ScriptRouteHandler(_loggerFactory, host, _environment, false), routePrefixToUse);
```

Comment:
Not sure if we want to do this or if we still want user to explicitly remove the prefix. Can't think of a reason why you'd want a catch-all route that has a prefix.